src/transformers/trainer.py

```diff
@@ -510,6 +516,10 @@ def __init__(
         ):
             self.place_model_on_device = False
 
+        self.tokenizer = processor if processor is not None else tokenizer
+        if processor is not None and hasattr(processor, "feature_extractor"):
+            tokenizer = processor.feature_extractor
```

Comment:
We should add a check here to ensure if the user has passed in both `tokenizer` and `processor`, to raise an error about you must only pass in one

Reply:
IMO it's ok to let the user pass both and have the `processor` take precedence (there's no harm in this for the user)

Reply:
However we never save their original tokenizer. This can lead to confusion down the road because their tokenizer is essentially never used. I'd rather guard this explicitly.;

Reply:
> IMO it's ok to let the user pass both and have the processor take precedence (there's no harm in this for the user)

I disagree here, it makes the behaviour ambiguous. In effect, this PR means we're deprecating the use of the `tokenizer` argument, so we should make it clear which argument is preferred and push the user towards that.